numba/typed/typedobjectutils.py

```diff
@@ -173,14 +173,18 @@ def build_wrapper(fn):
         context.call_conv.return_value(builder, intres)
 
     wrapfn = cgutils.get_or_insert_function(
-        module, wrapfnty, name='.numba_{}.{}_equal.wrap'.format(
-            context.fndesc.mangled_name, container_element_type))
+        module,
+        wrapfnty,
+        name=f".numba_{context.fndesc.mangled_name}.dict_key_equal.wrap",
+    )
```

Comment:
I was looking at https://github.com/numba/numba/pull/6402 and perhaps the `container_element_type` may be needed. Could you undo the changes related to removing the container_element_type? Just want to make sure we don't introduce any regression with this PR.

Reply:
Sure, addressed.